gridstatus/miso.py

```diff
@@ -679,8 +679,586 @@ def _get_generation_outages_data(
+    def get_binding_constraints_day_ahead_yearly_historical(
+        self,
+        year: int,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """Get the day-ahead binding constraints data from MISO for a given year.
+
+        Args:
+            year (int): Year
+            verbose (bool, optional): Verbosity. Defaults to False.
+
+        Returns:
+            pandas.DataFrame: Historical day-ahead binding constraints data
+        """
+        url = f"https://docs.misoenergy.org/marketreports/{year}_da_bc_HIST.csv"
+        logger.info(f"Downloading day-ahead binding constraints data from {url}")
+
+        data = pd.read_csv(url)
+        data["Interval End"] = pd.to_datetime(data["Market Date"]).dt.tz_localize(
+            self.default_timezone,
+        ) + pd.to_timedelta(data["Hour of Occurrence"], unit="h")
+        data["Interval Start"] = data["Interval End"] - pd.Timedelta(hours=1)
+
+        data = data.rename(
+            columns={
+                "Branch Name ( Branch Type / From CA / To CA )": "Branch Name",
+            },
+        )
+        data = data.rename(
+            columns={
+                "Constraint_ID": "Constraint ID",
+            },
+        )
+
+        return data[
+            [
+                "Interval Start",
+                "Interval End",
+                "Flowgate NERC ID",
+                "Constraint ID",
+                "Constraint Name",
+                "Branch Name",
+                "Contingency Description",
+                "Shadow Price",
+                "Constraint Description",
+                "Override",
+                "Curve Type",
+                "BP1",
+                "PC1",
+                "BP2",
+                "PC2",
+                "Reason",
+            ]
+        ]
+
+    @support_date_range(frequency="DAY_START")
+    def get_subregional_power_balance_constraints_day_ahead_hourly(
+        self,
+        date: str | pd.Timestamp,
+        end: str | pd.Timestamp = None,
+        verbose: bool = False,
+    ):
+        query_date = date - pd.Timedelta("1D")
+        url = f"https://docs.misoenergy.org/marketreports/{query_date.strftime('%Y%m%d')}_da_pbc.csv"
+        logger.info(
+            f"Downloading day-ahead subregional power balance constraints data from {url}",
+        )
+
+        data = pd.read_csv(
+            url,
+            skiprows=3,
+            index_col=False,
+            dtype={
+                "PRELIMINARY_SHADOW_PRICE": float,
+                "BP1": float,
+                "PC1": float,
+                "BP2": float,
+                "PC2": float,
+                "BP3": float,
+                "PC3": float,
+                "BP4": float,
+                "PC4": float,
+                "REASON": object,
+            },
+        )
+
+        # NOTE(kladar): The last row is a text disclaimer, and there is a leading space
+        # in the column names, so we clean it all up.
```

Comment:
Can we skip the last row with `skipfooter` in the `pd.read_csv` call? 

https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html

Reply:
```
/Users/kladar/GitHub/gridstatus/gridstatus/miso.py:865: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support skipfooter; you can avoid this warning by specifying engine='python'.
```
We can, but that changes to a different engine apparently which we need to explicitly call to avoid the warning. Oh pandas...

Not going to sweat it for now.

Reply:
Yeah, we specify the engine in a few places.

https://github.com/gridstatus/gridstatus/blob/74215686eb4d4aac1f8338ae5f03e2c7dc831467/gridstatus/miso.py#L294-L301